lua/telescope/_extensions/project.lua

```diff
@@ -0,0 +1,54 @@
+local has_telescope, telescope = pcall(require, 'telescope')
+
+if not has_telescope then
+	error('This plugins requires nvim-telescope/telescope.nvim')
+end
+
+local actions = require("telescope.actions")
+local finders = require("telescope.finders")
+local pickers = require("telescope.pickers")
+local sorters = require("telescope.sorters")
+local builtin = require("telescope.builtin")
+
+local project = function(opts)
+	opts = opts or {}
+
+	local project_dirs = {["nvim config"] = "~/.config/nvim"}
+	local projects = {}
+
+	local search_selected_project = function(selection)
+		print(project_dirs[selection])
+		local selected_directory = project_dirs[selection]
+		builtin.find_files({cwd = selected_directory})
+	end
+
+	local n = 0
+	for k in pairs(project_dirs) do
+		n = n + 1
+		projects[n] = k
+	end
+
+	local select_project = function()
+		pickers.new(opts, {
+			prompt_title = 'Select a project',
+			results_title = 'Projects',
+			finder = finders.new_table {results = projects},
+			sorter = sorters.get_generic_fuzzy_sorter(),
+			attach_mappings = function(prompt_bufnr, map)
+				local on_project_selected = function()
+					local selection = actions.get_selected_entry(prompt_bufnr)
+					actions.close(prompt_bufnr)
+					search_selected_project(selection.value)
+				end
+
+				map('i', '<CR>', on_project_selected)
+				map('n', '<CR>', on_project_selected)
```

Comment:
We can do `actions.goto_file_selection_edit:replace(actions.git_checkout)` rather than `map` that will replace the actual main action, no matter what key is mapped to it.

Reply:
Makes sense. I'll look into it

Reply:
Oh not `actions.git_checkout` here must be your function. like `on_project_selected`.

Reply:
I was wondering how this one was going to work! Thought maybe there was something I didn't know. Thanks